crates/egui-winit/src/lib.rs

```diff
@@ -163,22 +177,81 @@ impl State {
 
     /// Prepare for a new frame by extracting the accumulated input,
     /// as well as setting [the time](egui::RawInput::time) and [screen rectangle](egui::RawInput::screen_rect).
-    pub fn take_egui_input(&mut self, window: &winit::window::Window) -> egui::RawInput {
+    pub fn take_egui_input(
+        &mut self,
+        window: &winit::window::Window,
+        id_pair: ViewportIdPair,
+    ) -> egui::RawInput {
         let pixels_per_point = self.pixels_per_point();
 
         self.egui_input.time = Some(self.start_time.elapsed().as_secs_f64());
 
+        // TODO remove this in winit 0.29
+        // This hack make the window outer_position and size to be valid, X11 Only
+        // That was happending because winit get the window state before the compositor adds decorations!
+        #[cfg(feature = "x11")]
+        window.set_maximized(window.is_maximized());
```

Comment:
This caused an infinite loop on macOS. `x11` appears to be in the default features. Maybe we can change this to:

```suggestion
        // TODO remove this in winit 0.29
        // This hack make the window outer_position and size to be valid, X11 Only
        // That was happending because winit get the window state before the compositor adds decorations!
        #[cfg(all(feature = "x11", target_os = "linux"))]
        window.set_maximized(window.is_maximized());
```